data/account/participationRegistry.go

```diff
@@ -704,11 +716,16 @@ func (db *participationDB) Insert(record Participation) (id ParticipationID, err
 		Voting:            voting,
 		VRF:               vrf,
 	}
+	if record.StateProofSecrets != nil {
+		var stateProofPk StateProofPublicKey
+		copy(stateProofPk[:], record.StateProofSecrets.GetVerifier()[:])
```

Comment:
Same as above, I think you should stay consistent with how the voting the vrf are copied, and move this code to line 703.